Mage.Client/src/main/java/org/mage/card/arcane/Util.java

```diff
@@ -37,9 +37,13 @@ public Thread newThread(Runnable runnable) {
     }
 
     public static void broadcast(byte[] data, int port) throws IOException {
-        DatagramSocket socket = new DatagramSocket();
-        broadcast(socket, data, port, NetworkInterface.getNetworkInterfaces());
-        socket.close();
+        DatagramSocket socket = null;
+        try {
+            socket = new DatagramSocket();
+            broadcast(socket, data, port, NetworkInterface.getNetworkInterfaces());
+        } finally {
+            socket.close();
```

Comment:
`socket = new DatagramSocket();` must be outside from try-finally block... on failed `new Datagram` close with failed with null point exception.